core/src/main/java/org/apache/calcite/interpreter/Bindables.java

```diff
@@ -569,6 +633,56 @@ public Node implement(InterpreterImplementor implementor) {
     }
   }
 
+  /** Implementation of {@link org.apache.calcite.rel.core.Intersect} in
+   * bindable calling convention. */
+  public static class BindableIntersect extends Intersect implements BindableRel {
+    public BindableIntersect(RelOptCluster cluster, RelTraitSet traitSet,
+        List<RelNode> inputs, boolean all) {
+      super(cluster, traitSet, inputs, all);
+    }
+
+    public BindableIntersect copy(RelTraitSet traitSet, List<RelNode> inputs, boolean all) {
+      return new BindableIntersect(getCluster(), traitSet, inputs, all);
+    }
+
+    public Class<Object[]> getElementType() {
+      return Object[].class;
+    }
+
+    public Enumerable<Object[]> bind(DataContext dataContext) {
+      return help(dataContext, this);
+    }
+
+    public Node implement(InterpreterImplementor implementor) {
+      return new SetOpNode(implementor.compiler, this);
+    }
+  }
+
+  /** Implementation of {@link org.apache.calcite.rel.core.Minus} in
+   * bindable calling convention. */
+  public static class BindableMinus extends Minus implements BindableRel {
+    public BindableMinus(RelOptCluster cluster, RelTraitSet traitSet,
+        List<RelNode> inputs, boolean all) {
+      super(cluster, traitSet, inputs, all);
+    }
+
+    public BindableMinus copy(RelTraitSet traitSet, List<RelNode> inputs, boolean all) {
+      return new BindableMinus(getCluster(), traitSet, inputs, all);
```

Comment:
These Bindables should have some common abstraction.

Reply:
i think it is not , those Bindables should be return itself. Such as BindableMinus copy method return BindableMinus .

Reply:
> These Bindables should have some common abstraction.
Thanks for review, should we follow the LogicalUnion、LogicalIntersect、LogicalMinus copy method return itself ?